string first_node_name
string second_node_name
uint32 distance